gdx-pay-android-huawei/src/main/java/com/badlogic/gdx/pay/android/huawei/HuaweiPurchaseManager.java

```diff
@@ -224,29 +229,34 @@ public void onSuccess(PurchaseIntentResult result) {
                     // Obtain the payment result.
                     Status status = result.getStatus();
                     if (status.hasResolution()) {
-                        huaweiPurchaseManagerConfig.iapListener.onPurchaseResult(result);
+                        try {
+                            result.getStatus().startResolutionForResult(activity, PURCHASE_STATUS_RESULT_CODE);
+                        } catch (IntentSender.SendIntentException e) {
+                            huaweiPurchaseManagerConfig.observer.handlePurchaseError(e);
+                        }
                     }
                 }
             }).addOnFailureListener(new OnFailureListener() {
                 @Override
                 public void onFailure(Exception e) {
-                    if (e instanceof IapApiException) {
-                        IapApiException apiException = (IapApiException) e;
-                        huaweiPurchaseManagerConfig.iapListener.onIAPError(apiException);
-                    } else {
-                        huaweiPurchaseManagerConfig.iapListener.onError(e);
-                    }
-
                     huaweiPurchaseManagerConfig.observer.handlePurchaseError(e);
                 }
             });
         } else {
-            Exception e = new ProductInfoNotFoundException();
-            huaweiPurchaseManagerConfig.iapListener.onError(e);
-            huaweiPurchaseManagerConfig.observer.handlePurchaseError(e);
+            huaweiPurchaseManagerConfig.observer.handlePurchaseError(new ProductInfoNotFoundException());
         }
     }
 
+    public void handlePurchase(PurchaseResultInfo purchaseResultInfo) {
+        Transaction transaction = HuaweiPurchaseManagerUtils.
+                getTransactionFromPurchaseData(purchaseResultInfo.getInAppPurchaseData(), purchaseResultInfo.getInAppDataSignature());
+        this.huaweiPurchaseManagerConfig.observer.handlePurchase(transaction);
+    }
+
+    public void handlePurchaseError(String message, int code) {
+        this.huaweiPurchaseManagerConfig.observer.handlePurchaseError(new PurchaseError(message, code));
```

Comment:
clients won't be able to handle this purchase error, it is not in scope and they won't be able to use this code. So instead of introducing a new type of error, better use the generic GdxPayException with a specific message, or map the error codes to the existing subclasses of GdxPayExceptions. See the other implementations for examples.